ShoppingM8-be/src/main/java/edu/pw/shoppingm8/list/ListServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package edu.pw.shoppingm8.list;
+
+import org.springframework.stereotype.Service;
+
+import edu.pw.shoppingm8.authentication.AuthenticationService;
+import edu.pw.shoppingm8.list.api.dto.ListDto;
+import edu.pw.shoppingm8.list.exception.ListNotFoundException;
+import edu.pw.shoppingm8.user.User;
+import edu.pw.shoppingm8.user.UserService;
+
+@Service
+public class ListServiceImpl implements ListService {
+    private final ListRepository listRepository;
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    public ListServiceImpl(ListRepository listRepository,
+        UserService userService, AuthenticationService authenticationService) {
+        this.listRepository = listRepository;
+        this.userService = userService;
+        this.authenticationService = authenticationService;
+    }
+
+    @Override
+    public List create(ListDto listDto) {
+        User owner = authenticationService.getAuthenticatedUser();
+        List created = List.builder()
+            .name(listDto.getName())
+            .owner(owner).build();
+        return listRepository.save(created);
+    }
+
+    @Override 
+    public List update(List list, ListDto listDto) {
+        if (listDto.getName() != null)
+            list.setName(listDto.getName());
+        if (listDto.getOwnerId() != null) {
+            User owner = userService.getUser(listDto.getOwnerId());
```

Comment:
I don't think the owner field should be modifiable. App does not support transfering lists, unless we add it in scope.

Reply:
OK, I will change it